lib/spack/spack/spec.py

```diff
@@ -195,7 +195,7 @@
 )
 
 #: specfile format version. Must increase monotonically
-SPECFILE_FORMAT_VERSION = 5
+SPECFILE_FORMAT_VERSION = 6
```

Comment:
If the change is forward, and backward compatible, do we really need this bump?

Reply:
I think so, but only if I change it to only read the `direct` field if it's there?

Reply:
I think we shouldn't bump this number since:
1. It's a pain for users, and we just did it for compiler as deps
2. `direct` is only needed for abstract specs, which are not persisted
3. Currently the implementation doesn't rely on it for stuff that is sent across processes

We can probably get by with just a small modification to SpecfileV5. 

Note that we did something similar for #47351